src/OpenTelemetry.Extensions.Enrichment/TraceEnrichmentBag.cs

```diff
@@ -0,0 +1,36 @@
+// <copyright file="TraceEnrichmentBag.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+
+namespace OpenTelemetry.Extensions.Enrichment;
+
+#pragma warning disable CA1815 // Override equals and operator equals on value types
+public struct TraceEnrichmentBag
```

Comment:
Can we make this `public readonly struct`?

Reply:
Done